pkg/dataobj/planner/physical/plan.go

```diff
@@ -0,0 +1,297 @@
+package physical
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+)
+
+type NodeType uint32
+
+const (
+	NodeTypeDataObjScan NodeType = iota
+	NodeTypeSortMerge
+	NodeTypeProjection
+	NodeTypeFilter
+	NodeTypeLimit
+)
+
+func (t NodeType) String() string {
+	switch t {
+	case NodeTypeDataObjScan:
+		return "DataObjScan"
+	case NodeTypeSortMerge:
+		return "SortMerge"
+	case NodeTypeProjection:
+		return "Projection"
+	case NodeTypeFilter:
+		return "Filter"
+	case NodeTypeLimit:
+		return "Limit"
+	default:
+		return "Undefined"
+	}
+}
+
+// Node represents a single operation in a physical execution plan.
+// It defines the core interface that all physical plan nodes must implement.
+// Each node represents a specific operation like scanning, filtering, or
+// transforming data.
+// Nodes can be connected to form a directed acyclic graph (DAG) representing
+// the complete execution plan.
+type Node interface {
+	// Traversable accepts a node to be visited by the Visitor
+	Traversable
+	// ID returns a string that uniquely identifies a node in the plan
+	ID() string
+	// Type returns the node type
+	Type() NodeType
```

Comment:
Food for thought: will there ever be two nodes which return the same Type? If not, we might not need the `NodeType` type or this method.

Reply:
Theoretically yes, in practice no.

Reply:
The `NodeType` is also used for the string representation of the node (e.g. `DataObjScan`) when printed

Reply:
Ah, ok 👍 In some cases you could print `%T` to print the concrete Go type of the node, but that's not always the prettiest thing.